### PR TITLE
docs: add Tooltip documentation and examples

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Added `Tooltip` article and code examples.

### CODE REVIEW

File: articles/components/tooltip/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+The delay before tooltips appear can be configured separately for hover and keyboard focus.
+The delay before tooltips disappear once the pointer leaves the target element can also be separately configured.
+On blur, the tooltip is closed immediately to avoid confusion when focusing another element.
+
+[source,java]
+----
+// Global delay configuration:
+Tooltip.setDefaultFocusDelay(2000);
+Tooltip.setDefaultHoverDelay(1000);
+Tooltip.setDefaultHideDelay(1000);
```
Comment (tomivirkki) [inline articles/components/tooltip/index.asciidoc:83]: In Java, this is `TooltipConfiguration.setDefaultFocusDelay(...);`

File: src/main/java/com/vaadin/demo/component/tooltip/TooltipManual.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        textField.setPlaceholder("Search");
+        textField.setPrefixComponent(new Icon("lumo", "search"));
+        Button button = new Button(new Icon(VaadinIcon.INFO_CIRCLE));
+        button.addThemeVariants(ButtonVariant.LUMO_TERTIARY_INLINE, ButtonVariant.LUMO_ICON);
+        textField.setSuffixComponent(button);
+        // tag::snippet[]
+        textField.setTooltipText("Wrap in “quotes” for exact phrase");
+        Tooltip tooltip = textField.getTooltip().withManual(true);
+        button.addClickListener(event -> {
+            tooltip.setManual(!tooltip.isManual());
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/tooltip/TooltipManual.java:27]: This should open the tooltip instead of toggling the manual state

**Comment by tomivirkki** [comment]
Accidental removal of `package-lock.json`?

File: articles/components/tooltip/index.asciidoc (hunk 1 above)
Comment (web-padawan) [reply articles/components/tooltip/index.asciidoc]: Fixed.

File: src/main/java/com/vaadin/demo/component/tooltip/TooltipManual.java (hunk 2 above)
Comment (web-padawan) [reply src/main/java/com/vaadin/demo/component/tooltip/TooltipManual.java]: Fixed.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**